### PR TITLE
Changing import methods to better support pipx installs

### PR DESCRIPTION
**Pull request by SeanTConrad**
## Why are these changes needed?

Non-pipx installs won't work with some subprocess calls. This amends that.
Also added config flag for "run_git_stats". "run_git" was being overloaded for fetching repo and running git stats. When testing other parts of the code, it was very annoying to not be able to disable other scan modules.

## Checks

- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

## Summary by Sourcery

Add a new configuration option 'runGitStats' to allow independent control of git statistics gathering, and refactor subprocess calls to enhance compatibility with pipx installations.

New Features:
- Introduce a new configuration flag 'runGitStats' to separately control the execution of git statistics gathering, allowing more granular control over which scan modules are enabled.

Enhancements:
- Refactor the subprocess call for Semgrep scanning to improve compatibility with pipx installations by using a more direct method of invoking the scan command.

### CODE REVIEW

File: src/verinfast/agent.py [hunk 1]
```diff
@@ -434,90 +435,93 @@ def parseRepo(self, path: str, repo_name: str, branch: str = None):
...
-                            ], stderr=e,)
-                    except subprocess.CalledProcessError as e:
-                        output = e.output
-                        self.log(msg=output, tag="Scanning repository return", display=True)
+
+            findings_output_file = os.path.join(self.config.output_dir, repo_name + ".findings.json")
+            findings_error_file = os.path.join(self.config.output_dir, repo_name + ".findings.err")
+            if not self.config.dry:
+                self.log(msg=repo_name, tag="Scanning repository", display=True)
                 try:
```
[the remark this answers is not in the document]
Comment (SeanTConrad) [reply src/verinfast/agent.py:442]: SemGrep, as per its documenation, always calls SystemExit. We need to catch and stop it and there's no sense in logging something that happens every time.

File: src/verinfast/agent.py [hunk 2]
```diff
@@ -434,90 +435,93 @@ def parseRepo(self, path: str, repo_name: str, branch: str = None):
...
-                                "auto",
-                                "--json",
-                                "-o",
-                                findings_output_file,
-                            ], stderr=e,)
-                    except subprocess.CalledProcessError as e:
-                        output = e.output
-                        self.log(msg=output, tag="Scanning repository return", display=True)
+
+            findings_output_file = os.path.join(self.config.output_dir, repo_name + ".findings.json")
```
[the remark this answers is not in the document]
Comment (SeanTConrad) [reply src/verinfast/agent.py:438]: Semgrep quiet mode doesn't stop all output

File: src/verinfast/config.py [hunk 3]
```diff
@@ -167,6 +167,7 @@ class Config(printable):
     reportId: int = 0
     runDependencies: bool = True
     runGit: bool = True
```
Comment (aylusltd) [inline src/verinfast/config.py:169]: Don't mix PRs, this should be in a broader PR with tests and extra configs.

**Review by aylusltd — CHANGES_REQUESTED** [review]

File: src/verinfast/config.py (hunk 3 above)
Comment (SeanTConrad) [reply src/verinfast/config.py]: Removed

**Review by aylusltd — APPROVED** [review]

**Pull request merged**